src/xbot/common/injection/wpi_factories/RealWPIFactory.java

```diff
@@ -114,12 +125,16 @@ public XGyro getGyro() {
             // if we have a bad gyro.
 
             log.error("Could not create gyro! Returning a \"broken\" MockGyro instead.");
-
-            MockGyro brokenGyro = new MockGyro(new MockRobotIO());
-            brokenGyro.setIsBroken(true);
-            return brokenGyro;
+            return getBrokenGyro();
         }
     }
+    
+    private XGyro getBrokenGyro()
```

Comment:
Can we add an optional argument `isBroken` in the `MockGyro` constructor? This would be much more elegant and easy to understand at a glance as a one-liner.